utils/src/main/scala/za/co/absa/enceladus/utils/general/Section.scala

```diff
@@ -0,0 +1,288 @@
+
+/**
+  * Represents a section of a string defined by its starting index and length of the section.
+  * It supports negative indexes for start, denoting a position counted from end
+  * _Comparison_
+  * The class implements the `Ordered` trait
+  * Sections with negative start are BIGGER then the ones with positive start
+  * The section with smaller ABSOLUTE value of start is considered smaller, if starts equal the shorter section is the
+  * smaller
+  * NB! Interesting consequence of the ordering is, that if the sections are not overlapping and removal applied on
+  *  string from greatest to smallest one by one, the result is the same as removing all sections "at once"
+  * @param start the start position of the section, if negative the position is counted from the end
+  * @param length length of the section, cannot be negative
+  */
+case class Section(start: Int, length: Int) extends Ordered[Section] {
+
+  if (length < 0) {
+    throw new InvalidParameterException(s"'length; cannot be negative, $length given")
+  }
+
+  if ((start >= 0) && (start.toLong + length.toLong > Int.MaxValue)) {
+    throw new IndexOutOfBoundsException(s"start and length combination are grater then ${Int.MaxValue}")
+  }
+
+  override def compare(that: Section): Int = {
+    if (start == that.start) {
+      length.compare(that.length) //shorter is smaller
+    } else if ((start < 0) && (that.start >= 0)) {
+      1 // negative start is bigger then the one of positive+zero
+    } else if ((start >= 0) && (that.start < 0)) {
+      -1
+    } else if (start.abs < that.start.abs) {
+      -1
+    } else {
+      1
+    }
+  }
+
+  /**
+    * Converts the Section to actual indexes representing the section on the given string, if used in the substring
+    * function. The result respects the boundaries of the string, the indexes in the result not to cause OutOfBound exception
+    * For example Section(2,3) for string "Hello!" gives (2,5), for "abc" (2,3)
+    * @param forString  the string which the section would be applied to
+    * @return           tuple representing the beginIndex and endIndex parameters for the substring function
+    */
+  def toSubstringParameters(forString: String): (Int, Int) = {
+    val (realStart, after) = if (start >= 0) {
+        (start min forString.length, start + length)
+      } else {
+        val startIndex = forString.length + start
+        if (startIndex >= 0) { //
+          (startIndex, startIndex + length)
+        } else { // the distance from end is longer than the string itself
+          (0, length + startIndex max 0)
+        }
+      }
+    (realStart, after min forString.length)
+  }
+
+  /**
+    * The substring represented by this Section within the provided string
+    * Complementary to `remove`
+    * @param fromString the string to apply the section to
+    * @return           substring defined by this section
+    */
+  def extract(fromString: String): String = {
```

Comment:
```suggestion
  def extractFrom(string: String): String = {
```

Do a similar change the rest of methods as consulted at the meeting